File: Level-1/57.java
class Solution {
    public String solution(String X, String Y) {
        String answer = "";
        int[] x = new int[10];
        int[] y = new int[10];
        StringBuilder sb = new StringBuilder();
        
        for(int i = 0; i < X.length(); i++){
            int temp = X.charAt(i) - '0';
            x[temp]++;
        }
        
        for(int i = 0; i < Y.length(); i++){
            int temp = Y.charAt(i) - '0';
            y[temp]++;
        }
        
        for(int i = 9; i >= 0; i--){
            if(x[i] > 0 && y[i] > 0){
                if(x[i] == y[i]){
                    for(int j = 0; j < x[i]; j++){
                        sb.append(i);
                    }
                }else if(x[i] > y[i]){
                    for(int j = 0; j < y[i]; j++){
                        sb.append(i);
                    }
                }else{
                    for(int j = 0; j < x[i]; j++){
                        sb.append(i);
                    }
                }
            }
        }
        
        answer = sb.toString();
        
        if(answer.length() == 0) answer = "-1";
        if(answer.charAt(0) == '0') answer = "0";
         
        return answer;
    }
}

File: Level-1/57.js
function solution(X, Y) {
    let answer = '';
    X = X.split('');
    Y = Y.split('');
    
    for(let i = 9; i >= 0; i--) {
        const curX = X.filter((v) => Number(v) === i).length;
        const curY = Y.filter((v) => Number(v) === i).length;
        answer += String(i).repeat(Math.min(curX, curY));
    }
    
    if(answer === '') return '-1';
    if(Number(answer) === 0) return '0';
    
    return answer;
}
